swap/swap.go

```diff
@@ -542,7 +546,18 @@ func (s *Swap) getContractOwner(ctx context.Context, address common.Address) (co
 	return contr.Issuer(nil)
 }
 
-// Deploy deploys the Swap contract
+// NewInstanceAt creates a new instance of the chequebook contract at address and sets chequebookAddr
```

Comment:
I find this comment to be confusing. It doesn't really create a new instance of the contract but points to an existing one.

Reply:
isn't it an instance on the go side? i.e. a new `struct`?

Reply:
I would rename this to `BindToContractAt` or something similar. It is indeed very confusing, as it is a method on `Swap`, and it is not creating a new instance of `Swap`, and, as @ralph-pichler says, not even a new contract.

The comment should be changed accordingly

Reply:
I agree about this being confusing. 
I propose to change it to:

`NewInstanceAt creates a new instance of an already existing chequebook contract at address and sets chequebookAddr`

What do you think?

Reply:
@mortelli , it is indeed *only* a new instance on the go-side.
To be precise:
```golang
type simpleContract struct {
	instance *contract.SimpleSwap
}
```
, which is implementing the `Contract interface`
```golang
type Contract interface {
	// SubmitChequeBeneficiary submits a cheque to a given beneficiary
	SubmitChequeBeneficiary(opts *bind.TransactOpts, backend Backend, serial *big.Int, amount *big.Int, timeout *big.Int, ownerSig []byte) (*types.Receipt, error)
	// CashChequeBeneficiary cashes the cheque by the beneficiary
	CashChequeBeneficiary(auth *bind.TransactOpts, backend Backend, beneficiary common.Address, requestPayout *big.Int) (*types.Receipt, error)
	// ContractParams returns contract info (e.g. deployed address)
	ContractParams() *Params
	// Issuer returns the contract owner from the blockchain
	Issuer(opts *bind.CallOpts) (common.Address, error)
	// Cheques returns the last cheque for the given address
	Cheques(opts *bind.CallOpts, addr common.Address) (*ChequeResult, error)
}
```
(see file `contracts/swap.go`).

Reply:
since the instance exists in the blockchain but not on the go side, maybe use `reference` instead of `instance`; it seems talking about instances is a half-truth.

@ralph-pichler any suggestions for a better comment? what do you think of @Eknir's proposal?

Reply:
I would definitely not use `New`. `New` in go usually is used to create a new instance of the caller, not a new instance of some other object.

Maybe `BindToReferenceAt` or `BindToContractAt`, `ContractAt` but not `New...`

Reply:
Changed the function to `ContractAt` and updated the documentation to:
```
//ContractAt binds to an instance of an already existing chequebook contract at address and sets chequebookAddr
```

Reply:
Final resolution: as this call is deferring to `contract.InstanceAt`, I see no reason why we would name it anything other than that. 
Hence, I decided to change it to:
```golang
// InstanceAt creates a new instance of an already existing chequebook contract at address and sets chequebookAddr
func (s *Swap) InstanceAt(address common.Address, backend swap.Backend) (err error) {
	s.contract, err = contract.InstanceAt(address, backend)
	if err != nil {
		return err
	}
	s.setChequebookAddr(address)
	return nil
}
```
@holisticode , tagging you, as you already gave a thumbs-up on my previous proposed solution to make it `ContractAt`